DNN Platform/DotNetNuke.Web.Client.ResourceManager/ClientResourceController.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+namespace DotNetNuke.Web.Client.ResourceManager
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using DotNetNuke.Abstractions.Application;
+    using DotNetNuke.Abstractions.ClientResources;
+
+    /// <inheritdoc />
+    public class ClientResourceController : IClientResourceController
+    {
+        private readonly IHostSettings hostSettings;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ClientResourceController"/> class.
+        /// </summary>
+        /// <param name="hostSettings">The host settings.</param>
+        public ClientResourceController(IHostSettings hostSettings)
+        {
+            this.hostSettings = hostSettings;
+            this.RegisterPathNameAlias("SharedScripts", "~/Resources/Shared/Scripts/");
+        }
+
+        private List<IFontResource> Fonts { get; set; } = new List<IFontResource>();
+
+        private List<IScriptResource> Scripts { get; set; } = new List<IScriptResource>();
+
+        private List<IStylesheetResource> Stylesheets { get; set; } = new List<IStylesheetResource>();
+
+        private Dictionary<string, string> PathNameAliases { get; set; } = new Dictionary<string, string>();
+
+        private List<string> FontsToExclude { get; set; } = new List<string>();
```

Comment:
We don't need to preserve order here, right?
```suggestion
        private HashSet<string> FontsToExclude { get; set; } = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```